core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -204,6 +204,31 @@ public void set(PropertyKey key, Object value) {
     set(key, value, Source.RUNTIME);
   }
 
+  /**
+   * Sets the value for the appropriate key in the {@link Properties}.
+   *
+   * @param key the key to set
+   * @param value the value for the key
+   */
+  @java.lang.Deprecated
+  public void set(@Nonnull PropertyKey key, @Nonnull String value) {
+    checkArgument(!value.equals(""),
```

Comment:
This is the same as the old logic. Don't as me why. I want to ask too.

Reply:
This is basically a relaxed version of the `set` API a few lines down for backward compatibility. A recent change introduced type to property key and the old way of `set(key, "true")` is no longer valid. Because there are probably other developers directly calling these internal APIs in their code this change could break their code. So we are adding an API that would accept string and only log a warning. This will be removed in the next release.

Reply:
The initial author of this validation logic is zac, @ZacBlanco Would you please give some more context? Thanks!

Reply:
The reason this check was added was to prevent ambiguous behavior when configuring properties. There are a few of different cases that occur when calling this method.

- user passes null as value
- user passes empty string
- user passes a non-empty string

In the case where someone passes a null or empty string, they are usually trying to accomplish what the `unset` method does. However, unless the documentation is written clearly (and the user reads it), they may think otherwise. `null` and  empty string do not convey useful information about what the user might want from a property, and so we decided to disallow it.

I can see a few potential cases for desiring an empty string (e.g. when your property is a literal string value, such as an api endpoint, like `"/api/v1"` or something), but AFAIK most alluxio properties are not interpreted as strings, but some type of integer/boolean. More often than not it is useless to set `""`, and would likely lead to more errors, so it is disallowed in the first place.

Reply:
@ZacBlanco Thanks for your reply, it make me know more than before. But I still guess there are some scenarios user intent to set empty string for a existing property, something like set or clear exclude list.